lib/ClangImporter/ImportName.cpp

```diff
@@ -1729,9 +1729,16 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
 
 /// Returns true if it is expected that the macro is ignored.
 static bool shouldIgnoreMacro(StringRef name, const clang::MacroInfo *macro) {
-  // Ignore include guards.
-  if (macro->isUsedForHeaderGuard())
-    return true;
+  // Ignore include guards. Try not to ignore definitions of useful constants,
+  // which may end up looking like include guards.
+  if (macro->isUsedForHeaderGuard()
+      && macro->getNumTokens() == 1
+      && macro->tokens()[0].isLiteral()) {
+      auto* tok = macro->tokens()[0].getLiteralData();
```

Comment:
Nitpick: We try to only use `auto` when the type is meaningless or obvious from the right-hand expression; in this case explicitly saying `const char *` is probably more appropriate. Also, the star should be attached to the variable name to match C's parsing rules.

Reply:
I got rid of that `auto*` in the newer version. I'm still using `auto` though because I think the result type of the array access is clear.